azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java

```diff
@@ -426,6 +431,57 @@ public void testPreparingFlowEvent() throws Exception {
             .get(KubernetesContainerizedImpl.DEFAULT_AZKABAN_BASE_IMAGE_NAME).getVersion());
   }
 
+  /**
+   * Test merging of flow properties and flor params.
+   * @throws Exception
+   */
+  @Test
+  public void testFlowPropertyAndParamsMerge() throws Exception {
+    final ExecutableFlow flow = createTestFlow();
+    flow.setExecutionId(3);
+    final Props flowProps = new Props();
+    flowProps.put("param.override.image.version", "1.2.3");
+    flowProps.put("regular.param", "4.5.6"); // Should be filtered out.
+    final Map<String, Props> propsMap = new HashMap<>();
+    propsMap.put("test", flowProps);
+    when(this.projectLoader.fetchProjectProperties(flow.getProjectId(), flow.getVersion())).thenReturn(propsMap);
+    //when(FlowLoaderUtils.loadPropsFromYamlFile(this.projectLoader, flow, null)).thenReturn(flowProps);
+    //doReturn(flowProps).when(FlowLoaderUtils.loadPropsFromYamlFile(this.projectLoader, flow, null));
+    final ExecutionOptions executionOptions = new ExecutionOptions();
+    flow.setExecutionOptions(executionOptions);
+    final Map<String, String> flowParams = flow.getExecutionOptions().getFlowParameters();
+    Assert.assertEquals(0, flowParams.size());
+    final Props mergedFlowPropsAndParams = flow.getFlowPropsAndParams(this.projectLoader);
+    Assert.assertEquals(1, mergedFlowPropsAndParams.size());
+    Assert.assertTrue(mergedFlowPropsAndParams.containsKey("image.version"));
+    Assert.assertEquals("1.2.3", mergedFlowPropsAndParams.get("image.version"));
+  }
+
+  /**
+   * Test merging of flow properties and flow params where flor params override the flow property.
+   * @throws Exception
+   */
+  @Test
+  public void testFlowPropertyAndParamsMergeWithOverwrite() throws Exception {
```

Comment:
Can you please cover usecase for Flow 2.0 as well?

Reply:
Flow2.0 is very tricky to cover as there are multiple steps in getting the props file compared to flow 1.0
Is it okay if I do it as a followup?

Reply:
Sure.